docs/Derive.md

```diff
@@ -0,0 +1,107 @@
+# Derive
+
+`derive` is a mechanism that automatically determines how to implement
+interfaces for datatypes based on their members. It also allows you to write
+your own rules for `derive`, called a `deriver`, but most users will likely
+not use that feature.
+
+If you’d like to learn how to `derive` interfaces for your types, read the
+first section of this document. If you’d like to provide a deriver for an
```

Comment:
Maybe link to the sections here using heading anchors?` [first section](#i-using-derive)`